test/e2e/node_feature_discovery_test.go

```diff
@@ -899,6 +926,39 @@ denyLabelNs: []
 
 })
 
+// simplyAndStupidRetryLoop is a simple and stupid re-try loop
+func simplyAndStupidRetryLoop(poll func() error) error {
+	var err error
+	for retry := 0; retry < 3; retry++ {
+		if err = poll(); err == nil {
+			return nil
+		}
+		time.Sleep(1 * time.Second)
+	}
+	return err
+}
+
+// waitForCapacity waits for the capacity to be updated in the node status
+func waitForCapacity(cli clientset.Interface, expected map[corev1.ResourceName]resourcev1.Quantity) error {
```

Comment:
With the current test data we also want to verify that no extra ERs (those ones that should be denied) are created. You can check what [checkForNodeLabels()](https://github.com/kubernetes-sigs/node-feature-discovery/blob/60f052f086154fd8f7eb4bd24208bbfb32c40578/test/e2e/node_feature_discovery_test.go#L933) does

Would be smth like:

```suggestion
func waitForCapacity(cli clientset.Interface, expectedNewERs map[corev1.ResourceName]resourcev1.Quantity, oldNodes []corev1.Node) error {
```